src/lib/run_environment.ml

```diff
@@ -94,10 +94,37 @@ module Tool = struct
   end
 end
 
+module Make_fun = struct
+  module Requirement = struct
+    type t = [
+      | `Processors of int
+      | `Internet_access
+      | `Memory of [ `GB of float | `Decent | `Big ]
+      | `Quick_run
```

Comment:
But then why would any job "not" be scheduled.

It's a `Machine.t`'s decision to say "I have a big, desert, login node, I can run some quick stuff here".